web_programming/current_stock_price.py

```diff
@@ -3,12 +3,20 @@
 
 
 def stock_price(symbol: str = "AAPL") -> str:
-    url = f"https://in.finance.yahoo.com/quote/{symbol}?s={symbol}"
-    soup = BeautifulSoup(requests.get(url).text, "html.parser")
-    class_ = "My(6px) Pos(r) smartphone_Mt(6px)"
-    return soup.find("div", class_=class_).find("span").text
+    url = f"https://finance.yahoo.com/quote/{symbol}?p={symbol}"
+    yahoo_finance_source = requests.get(url, headers={"USER-AGENT": "Mozilla/5.0"}).text
+    soup = BeautifulSoup(yahoo_finance_source, "html.parser")
+    specific_fin_streamer_tag = soup.find("fin-streamer", {"data-test": "qsp-price"})
 
+    if specific_fin_streamer_tag:
+        text = specific_fin_streamer_tag.get_text()
+        return text
+    else:
+        print("No <fin-streamer> tag with the specified data-test attribute found.")
+    return "Not Found"
```

Comment:
```suggestion
    if specific_fin_streamer_tag:
        text = specific_fin_streamer_tag.get_text()
        return text
    
    print("No <fin-streamer> tag with the specified data-test attribute found.")
    return "Not Found"
```

Also, why both print and return? Why not just return the error message? Or print the error message and return None?

Reply:
Thanks for the suggestion. Cannot return None as that would through an error in string formatting. But returning the error looks valid. Doing the change for that.